### PR TITLE
Add support for foreign key actions

### PR DESCRIPTION
**Pull request by toga4**
This PR adds support for [foreign key actions](https://cloud.google.com/spanner/docs/foreign-keys/overview?hl=en#how-to-define-foreign-key-action)

### CODE REVIEW

File: ast/ast.go [hunk 1]
```diff
@@ -1490,17 +1490,19 @@ type TableConstraint struct {
...
-	Foreign token.Pos // position of "FOREIGN" keyword
-	Rparen  token.Pos // position of ")" after reference columns
+	Foreign     token.Pos // position of "FOREIGN" keyword
+	Rparen      token.Pos // position of ")" after reference columns
+	OnDeleteEnd token.Pos // end position of ON DELETE clause
 
 	Columns          []*Ident
 	ReferenceTable   *Ident
 	ReferenceColumns []*Ident // len(ReferenceColumns) > 0
+	OnDelete         OnDeleteAction
```
Comment (makenowjust) [inline ast/ast.go:1505]:
```suggestion
	OnDelete         OnDeleteAction // optional
```

Comment (makenowjust) [reply ast/ast.go:1505]: Maybe `make fmt` is needed.

File: ast/ast.go [hunk 2]
```diff
@@ -1490,17 +1490,19 @@ type TableConstraint struct {
 
 // ForeignKey is foreign key specifier in CREATE TABLE and ALTER TABLE.
 //
-//	FOREIGN KEY ({{.ColumnNames | sqlJoin ","}}) REFERENCES {{.ReferenceTable}} ({{.ReferenceColumns | sqlJoin ","}})
+//	FOREIGN KEY ({{.ColumnNames | sqlJoin ","}}) REFERENCES {{.ReferenceTable}} ({{.ReferenceColumns | sqlJoin ","}}) {{.OnDelete}}
 type ForeignKey struct {
 	// pos = Foreign
 	// end = Rparen + 1
```
Comment (makenowjust) [inline ast/ast.go:1496]:
```suggestion
	// end = OnDeleteEnd || Rparen + 1
```

**Review by makenowjust — COMMENTED** [review]
Please fix some minor doc errors.

File: ast/ast.go (hunk 1 above)
Comment (toga4) [reply ast/ast.go]: 19d16af

File: ast/ast.go (hunk 2 above)
Comment (toga4) [reply ast/ast.go]: 19d16af

**Comment by toga4** [comment]
@makenowjust Thank you for review. Now fixed.

**Comment by makenowjust** [comment]
@toga4 Thank you!

**Pull request merged**